### PR TITLE
Prepend pip commands with activating env

### PR DESCRIPTION
**Pull request by soapy1**
addresses https://github.com/conda/conda/issues/8187

### CODE REVIEW

**Comment by msarahan** [comment]
Looks good to me.  Once the tests are in order, I think this can go in.  @jjhelmus please take a look, since you wrote the original functionality here.

**Comment by jjhelmus** [comment]
cf: #8229

File: conda_env/installers/pip.py [hunk 1]
```diff
@@ -56,6 +64,8 @@ def _pip_install_via_requirements(prefix, specs, args, *_, **kwargs):
         # So we delete the temporary file in a finally block.
         if requirements is not None and op.isfile(requirements.name):
             os.remove(requirements.name)
+        if script_caller is not None:
+            os.remove(script_caller)
```
Comment (jjhelmus) [inline conda_env/installers/pip.py]: As @kalefranz mentioned in #8229 this should be `rm_rf` not `os.remove` unless I am missing something.  If you want to change the call in line ~1065 of `link.py` that would be great.

File: conda/core/link.py [hunk 2]
```diff
@@ -1003,28 +1001,21 @@ def run_script(prefix, prec, action='post-link', env_prefix=None, activate=False
...
         shell_path = 'sh' if 'bsd' in sys.platform else 'bash'
         if activate:
-            conda_exe = env.get("CONDA_EXE", abspath(join(context.root_prefix, 'bin', 'conda')))
-            with tempfile.NamedTemporaryFile(mode='w', prefix=tmp_prefix, delete=False) as fh:
-                fh.write("eval \"$(\"{0}\" \"shell.posix\" \"hook\")\"\n".format(conda_exe)),
-                fh.write("conda activate \"{0}\"\n".format(prefix)),
-                fh.write("source \"{}\"\n".format(path))
-                script_caller = fh.name
-            command_args = [shell_path, "-x", script_caller]
+            command = ". \"{}\"\n".format(path)
```
Comment (jjhelmus) [inline conda/core/link.py]: Calling path directly and not souring is correct here.  Thanks you for correcting this mistake from #8229.

File: conda/utils.py [hunk 3]
```diff
@@ -254,3 +257,30 @@ def sys_prefix_unfollowed():
...
         return sys.prefix
     return unfollowed
+
+
+def wrap_subprocess_call(on_win, root_prefix, prefix, command):
+    env = environ.copy()
+    tmp_prefix = abspath(join(prefix, '.tmp'))
+    script_caller = None
+    if on_win:
+        comspec = environ[str('COMSPEC')]
```
Comment (jjhelmus) [inline conda/utils.py]: This environment lookup can fail but perhaps allowing the exception to propagate is the best option here.

**Comment by soapy1** [comment]
Looks like the python 2.7 tests are hitting the 1hr limit on appveyor

**Pull request merged**